compiler/src/dotty/tools/dotc/core/TypeErasure.scala

```diff
@@ -699,11 +746,12 @@ class TypeErasure(sourceLanguage: SourceLanguage, semiEraseVCs: Boolean, isConst
       }
     case _: ErrorType | JavaArrayType(_) =>
       tp
-    case tp: WildcardType if wildcardOK =>
+    case tp: WildcardType =>
+      assert(inSigName, i"Cannot erase wildcard type $tp")
+      null
+    case tp if (tp `eq` NoType) || (tp `eq` NoPrefix) => // Why is this check different?
       tp
-    case tp if (tp `eq` NoType) || (tp `eq` NoPrefix) =>
-      tp
-  }
+  ).ensuring(etp => etp != null || inSigName)
```

Comment:
`ensuring` is not inlined and therefore more expensive than an `assert`. Since `Erasure.apply` is hot, maybe use an assert instead?